pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -543,6 +544,20 @@ func (cr *PerconaXtraDBCluster) CompareVersionWith(version string) int {
 	return cr.Version().Compare(v.Must(v.NewVersion(version)))
 }
 
+// ConfigHasKey check if cr.Spec.PXC.Configuration has given key in given section
+func (cr *PerconaXtraDBCluster) ConfigHasKey(section, key string) (bool, error) {
+	file, err := ini.LoadSources(ini.LoadOptions{AllowBooleanKeys: true}, []byte(cr.Spec.PXC.Configuration))
+	if err != nil {
+		return false, err
+	}
+	s, err := file.GetSection(section)
+	if err != nil {
+		return false, err
+	}
```

Comment:
As I've seen in code `file.GetSection()` will return an error in the case of the section doesn't exist (https://github.com/percona/percona-xtradb-cluster-operator/blob/c8a79fbb2deffee71f4c2789afadf151fb33080e/vendor/github.com/go-ini/ini/ini.go#L296). It shouldn't be an error for the user.

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/548/commits/e21121a34ed2ef556c8e414635df5cbff52fd5de